### PR TITLE
Added option to hide already-known items from the mouseover tooltip.

### PR DESCRIPTION
**Pull request by dbowden713**
Implementing feature for issue/request posted [here](https://github.com/SacredDuckwhale/Rarity/issues/201).

Created a new DB variable `hideKnownItemsInTooltip` for tracking the new option, which is added to the World Tooltip Options. This option is checked before Rarity adds any info to the tooltip where an item could be already known.

Tested as many types of drops as I could, but there may be a few things left to check.

Note: doesn't hide achievement info.

Currently works for: 
- rare drops
- one-time drops
- zone drops
- drops from items (like boxes)

New option:
option

### CODE REVIEW

File: Core/GUI.lua [hunk 1]
```diff
@@ -1881,40 +1881,44 @@ _G.GameTooltip:HookScript(
...
-								attemptText = ""
-							end
-							GameTooltip:AddLine(
-								colorize(
-									(not rarityAdded and L["Rarity: "] .. (R.db.profile.blankLineAfterRarity and "\n" or "") or "") ..
-										(itemLink or itemName or v.name) .. attemptText,
-									yellow
+							if v.known and R.db.profile.hideKnownItemsInTooltip then
+								GameTooltip:Show()
+							else
```
Comment (dbowden713) [inline Core/GUI.lua:1886]: This is the main change. Everything else was just moved inside the `else` block.

**Comment by SacredDuckwhale** [comment]
Thank you! Could you rebase your branch to get rid of the unrelated merge commit (https://github.com/SacredDuckwhale/Rarity/pull/208/commits/3d51715c9623232d689eac2c3c3ba073b16bf624)?

I tend to use rebase when updating locally to avoid unnecessary merge commits that will persist in history.

I'll take a look at the rest ASAP, but I think simply forking the latest master and applying your changes is the easiest way to get this done.

**Comment by dbowden713** [comment]
Sorry about that! I've squashed everything into one commit and it seems to be cleaned up now. If I need to do anything else on my end just let me know.

**Comment by SacredDuckwhale** [comment]
The only thing that strikes me is the commit description still saying ``Fix blank line appearing in tooltips with already-known items hidden.`` :)

**Comment by dbowden713** [comment]
There we go!

**Comment by SacredDuckwhale** [comment]
Thanks again or your contribution! I'll add this to the latest alpha.

**Pull request merged**